crates/miden-lib/src/account/faucets/mod.rs

```diff
@@ -223,6 +219,236 @@ impl TryFrom<&Account> for BasicFungibleFaucet {
     }
 }
 
+// NETWORK FUNGIBLE FAUCET ACCOUNT COMPONENT
+// ================================================================================================
+
+// Initialize the digest of the `distribute` procedure of the Network Fungible Faucet only once.
+static NETWORK_FUNGIBLE_FAUCET_DISTRIBUTE: LazyLock<Word> = LazyLock::new(|| {
+    let distribute_proc_name = QualifiedProcedureName::new(
+        Default::default(),
+        ProcedureName::new(NetworkFungibleFaucet::DISTRIBUTE_PROC_NAME)
+            .expect("failed to create name for 'distribute' procedure"),
+    );
+    network_fungible_faucet_library()
+        .get_procedure_root_by_name(distribute_proc_name)
+        .expect("Network Fungible Faucet should contain 'distribute' procedure")
+});
+
+// Initialize the digest of the `burn` procedure of the Network Fungible Faucet only once.
+static NETWORK_FUNGIBLE_FAUCET_BURN: LazyLock<Word> = LazyLock::new(|| {
+    let burn_proc_name = QualifiedProcedureName::new(
+        Default::default(),
+        ProcedureName::new(NetworkFungibleFaucet::BURN_PROC_NAME)
+            .expect("failed to create name for 'burn' procedure"),
+    );
+    network_fungible_faucet_library()
+        .get_procedure_root_by_name(burn_proc_name)
+        .expect("Network Fungible Faucet should contain 'burn' procedure")
+});
+
+/// An [`AccountComponent`] implementing a network fungible faucet.
+///
+/// It reexports the procedures from `miden::contracts::faucets::basic_fungible`. When linking
+/// against this component, the `miden` library (i.e. [`MidenLib`](crate::MidenLib)) must be
+/// available to the assembler which is the case when using
+/// [`TransactionKernel::assembler()`][kasm]. The procedures of this component are:
+/// - `distribute`, which mints an assets and create a note for the provided recipient.
+/// - `burn`, which burns the provided asset.
+///
+/// `distribute` requires authentication while `burn` does not require authentication and can be
+/// called by anyone. Thus, this component must be combined with a component providing
+/// authentication.
+///
+/// This component supports accounts of type [`AccountType::FungibleFaucet`].
+///
+/// Unlike [`BasicFungibleFaucet`], this component uses two storage slots:
+/// - First slot: Token metadata `[max_supply, decimals, token_symbol, 0]`
+/// - Second slot: Creator account ID as a single Word
+///
+/// [kasm]: crate::transaction::TransactionKernel::assembler
+pub struct NetworkFungibleFaucet {
+    symbol: TokenSymbol,
+    decimals: u8,
+    max_supply: Felt,
+    creator_account_id: Word,
```

Comment:
nit: I'd probably name this `owner_account_id`. We may add functionality in the future that would allow changing owners - and so, the owner may not be the same entity as the creator.